testResources/testUtils.ts

```diff
@@ -14,6 +14,37 @@ import { expect } from "chai";
 
 const SPEC_PROFILE_PATH = path.join(__dirname, "raml/mercury/mercury.raml");
 
+const LIBRARY_DIR = `file://${__dirname}/../resources/lint/lib`;
+
+const BASE_CUSTOM_PROFILE = {
+  profile: "test_profile",
+  uses: {
+    "mercury-standards": `${LIBRARY_DIR}/mercury-standards.yaml`,
+    "slas-standards": `${LIBRARY_DIR}/slas-standards.yaml`,
+  },
+  extends: "RAML",
+  disabled: ["amf-parser.WebAPI-mediaType-datatype"],
+};
+
+export function generateValidationRules(lib: string, rules: string[]): {} {
+  const myValidations: { [key: string]: string } = {};
+  rules.forEach((rule) => (myValidations[rule] = `${lib}.${rule}`));
+  return {
+    violation: rules,
+    validations: myValidations,
+  };
+}
```

Comment:
```suggestion
export function generateValidationRules<T extends string>(lib: string, rules: T[]): { violation: T[]; validations: { [key: T]: string } } {
  const myValidations: { [key: T]: string } = {};
  rules.forEach((rule) => { myValidations[rule] = `${lib}.${rule}`; });
  return {
    violation: rules,
    validations: myValidations,
  };
}
```
The type `{}` means an object with no properties, so trying to access `generateValidationRules(lib, rules).violation` will result in a type error.